Revit_Core_Engine/Query/CeilingPattern.cs

```diff
@@ -68,6 +68,22 @@ public static List<BH.oM.Geometry.Line> CeilingPattern(this Ceiling ceiling, Rev
 
             double rotation;
             XYZ alignment = ceiling.CeilingPatternAlignment(material, settings, out rotation);
+            if (alignment == null)
+            {
+                // Try getting pattern of the main painted material if any
+                var paintedMaterialIds = ceiling.GetMaterialIds(true);
+                if (paintedMaterialIds.Count > 0)
+                {
+                    var mainPaintedId = paintedMaterialIds.Aggregate((id1, id2) => ceiling.GetMaterialArea(id1, true) > ceiling.GetMaterialArea(id2, true) ? id1 : id2);
```

Comment:
What if there is more than 2 painted materials? Wouldn't it be better to combine `OrderByDescending` and `First` Linq queries here?

Reply:
Actually, I believe Aggregate here works with any number of materials. It will compare the first 2 materials, get one that is greater in area, compare that to the 3rd material, get the bigger one, and then continue like that till the end of the item list.

Reply:
Right, good point 👍